src/main/java/games/strategy/util/Version.java

```diff
@@ -131,42 +135,41 @@ public int hashCode() {
   }
 
   @Override
-  public int compareTo(final Version o) {
-    return o == null ? -1 : compareTo(o, false);
+  public int compareTo(final Version other) {
+    checkNotNull(other);
+
+    return compareTo(other, false);
   }
 
   private int compareTo(final Version other, final boolean ignoreMicro) {
```

Comment:
I really think this overload is a code smell.
All calls to this method with ignoreMicro = true can be replaced with a call to isCompatible (and optionally, additionally isGreaterThan/isLessThan if the call does not compare against 0)

Reply:
I agree that this is a smell.

There's only one call that will invoke this method with `ignoreMicro` equal to `true`, and that's in `GameParser#parseMinimumEngineVersionNumber()`:

```java
private void parseMinimumEngineVersionNumber(final Node minimumVersion) throws EngineVersionException {
  if (minimumVersion == null) {
    return;
  }
  final Version mapCompatibleWithTripleaVersion =
      new Version(((Element) minimumVersion).getAttribute("minimumVersion"));
  if (mapCompatibleWithTripleaVersion.isGreaterThan(ClientContext.engineVersion(), true)) {
    throw new EngineVersionException(
        String.format("Current engine version: %s, is not compatible with version: %s, required by map: %s",
            ClientContext.engineVersion(),
            mapCompatibleWithTripleaVersion.toString(),
            data.getGameName()));
  }
}
```

I originally thought that the `isGreaterThan(Version, boolean)` call could be replaced with `isCompatible()`, but now I'm not positive.  The above check will be `false` if `mapCompatibleWithTripleaVersion` is less than the current engine version.  However, if `!isCompatible()` was used, the expression value would change to `true`.

There seem to be two distinct types of compatibility: engine-to-engine compatibility for communication (governed by `isCompatible()`) and map-to-engine compatibility for playing a specific map (governed by `isGreaterThan(Version, boolean)`).